### PR TITLE
Improving Docker Image

### PR DESCRIPTION
**Pull request by mohammedalqadi**
This is interesting project which made me want to help a little, any feedback will help :)

An improvement for docker build and image by using lsiobase/python from linuxserver.io Team, for smaller image size and the ability to map custom User PUID/PGID and an option to change server port.

reference from:
https://github.com/linuxserver/docker-mods
https://github.com/linuxserver/docker-baseimage-alpine-python

### CODE REVIEW

**Comment by benbusby** [comment]
Looks good to me at first glance, thanks for your contribution! I should have a chance later to do a more thorough review of the changes and let you know if any modifications are needed.

**Comment by mohammedalqadi** [comment]
Of course more testing required to make sure everything is okay, And I added slight change to variable validation.

**Comment by sUyMur** [comment]
Had to change from "FROM lsiobase/python:3.11" to "FROM lsiobase/python:arm32v7-3.11" since I'm on Raspberry Pi.
Maybe there is a possibility to implement build for x86 / armv7 etc.?

**Comment by mohammedalqadi** [comment]
> Works for me - only had to change from "FROM lsiobase/python:3.11" to "FROM lsiobase/python:arm32v7-3.11" because I'm on Raspberry Pi.
> 
> Maybe there is a possibility to implement builds for x86 / armv7 etc.?

I Updated the files to match the latest changes from the project, And there is Dockerfile for arm32 and arm64 if like to test.

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,67 @@
...
+	.build-dependencies && \
+ rm -rf \
+	/tmp/*
+
+# set config location
+ENV CONFIG_VOLUME="/config"
 
-CMD ["./whoogle-search"]
+# add local files
+COPY root/ /
```
Comment (mohammedalqadi) [inline Dockerfile:63]: We can change and remove the local files copy command and it will be pulled directly from GitHub.

File: Dockerfile [hunk 2]
```diff
@@ -1,13 +1,67 @@
...
+ curl -o \
+      /tmp/whooglesearch.tar.gz -L \
+      "https://github.com/benbusby/whoogle-search/archive/${WHOOGLESEARCH_RELEASE}.tar.gz" && \
+ tar xf \
+      /tmp/whooglesearch.tar.gz -C \
+      /tmp/whooglesearch --strip-components=1 && \
+ echo "**** install pip packages ****" && \
+ pip3 install --no-cache-dir -r \
+      /tmp/whooglesearch/requirements.txt && \
+      cp -r /tmp/whooglesearch/app / && \
```
Comment (mohammedalqadi) [inline Dockerfile:52]: Copy root folder from project directly by changing to: 
cp -r /tmp/whooglesearch/app / && cp -r /tmp/whooglesearch/root /app/  && \

**Comment by sUyMur** [comment]
> > Works for me - only had to change from "FROM lsiobase/python:3.11" to "FROM lsiobase/python:arm32v7-3.11" because I'm on Raspberry Pi.
> > Maybe there is a possibility to implement builds for x86 / armv7 etc.?
> 
> I Updated the files to match the latest changes from the project, And there is Dockerfile for arm32 and arm64 if like to test.

I only have a 32bit OS installed (DietPi) and can confirm that I can build off of "Dockerfile.armhf".

Commands to reproduce:
`git clone https://github.com/mohammedalqadi/whoogle-search`
`mv Dockerfile Dockerfile_original && mv Dockerfile.armhf Dockerfile`
`docker build --tag whooglesearch_arm32test:1.0 .`
Results in:
> Successfully built 6c2a15c8208f
> Successfully tagged whooglesearch_arm32test:1.0

docker-compose up -d also works:
`Creating whoogle-search ... done`

Used docker-compose.yml:
```
version: "3"

services:

     whoogle-search:
         container_name: whoogle-search_arm32test
         hostname: whoogle-search
         image: "whooglesearch_arm32test:1.0"
         restart: always
         ports:
           - "5000:5000"
```

**Comment by xoseperez** [comment]
Thanks for this, it worked almost out of the box.
I had to add `libxml2-dev` and `libxslt-dev` on a fresh Raspbian stretch to make it work...

**Comment by benbusby** [comment]
Just wanted to check in and say I haven't forgotten about this PR, I'm in the middle of a pretty large refactoring of the project that I'd like to get merged in first before doing a more thorough review to make sure it all lines up well with everything I'm changing. Hopefully I can look this over more in depth soon!

**Comment by mohammedalqadi** [comment]
> Thanks for this, it worked almost out of the box.
> I had to add `libxml2-dev` and `libxslt-dev` on a fresh Raspbian stretch to make it work...

Happy to help and Thank you for testing, I've added the missing packages libxml2-dev and libxslt-dev now it should work as intended.

**Comment by mohammedalqadi** [comment]
> Just wanted to check in and say I haven't forgotten about this PR, I'm in the middle of a pretty large refactoring of the project that I'd like to get merged in first before doing a more thorough review to make sure it all lines up well with everything I'm changing. Hopefully I can look this over more in depth soon!

I am excited to see the changes and improvements and help in any way possible :)

**Pull request merged**